File: g.vb
Module GoldenRule
    Function Calculate(to_val As Integer) As Double
        Const NUMBER_ONE As Integer = 1
        Dim i As Integer = NUMBER_ONE
        Dim j As Integer = NUMBER_ONE
        Dim tmp As Integer = 0
        For val As Integer = 1 To to_val
            tmp = i + j
            i = j
            j = tmp
        Next
        Return j / i
    End Function

    Sub Main()
        Console.WriteLine(Calculate(24).ToString)
    End Sub
End Module

File: g.cs
using System;

namespace GoldenRule
{
    class Program
    {
        static double Calculate(int to_val)
        {
            const int NUMBER_ONE = 1;
            int i = NUMBER_ONE;
            int j = NUMBER_ONE;
            int tmp = 0;
            for (int val = 0; val < to_val; val++)
            {
                tmp = i + j;
                i = j;
                j = tmp;
            }
            return (double)j / i;
        }

        static void Main(string[] args)
        {
            Console.WriteLine(Calculate(24).ToString());
        }
    }
}
